x/lockup/keeper/lock.go

```diff
@@ -393,9 +410,14 @@ func (k Keeper) BeginUnlock(ctx sdk.Context, lock types.PeriodLock) error {
 		return err
 	}
 
-	// remove from accumulation store
+	// remove from not unlocking accumulation store
 	for _, coin := range lock.Coins {
-		k.accumulationStore(ctx, coin.Denom).Remove(accumulationKey(lock.Duration, lock.ID))
+		k.accumulationStore(ctx, false, coin.Denom).Remove(accumulationKey(lock.Duration, lock.ID))
+	}
+
+	// add to unlocking accumulation store
+	for _, coin := range lock.Coins {
+		k.accumulationStore(ctx, true, coin.Denom).Set(combineKeys(getTimeKey(lock.EndTime), sdk.Uint64ToBigEndian(lock.ID)), coin.Amount)
```

Comment:
Why is the lock.ID included here?

Reply:
Same architecture for duration basis and followed it.
```
	// add to accumulation store
	for _, coin := range lock.Coins {
		k.accumulationStore(ctx, false, coin.Denom).Set(accumulationKey(lock.Duration, lock.ID), coin.Amount)
	}
```

Reply:
I see, lets keep this for now, and make an issue to remove it in both cases later

Reply:
Ok